src/utils.js

```diff
@@ -60,3 +62,33 @@ export function recalculateActive (terminalsSet, active) {
 		t.emitter.emit('did-change-title')
 	})
 }
+
+// finds the default shell start commmand
+async function getShellStartCommand () {
+	let shellStartCommand
+	if (process.platform === 'win32') {
+		// Windows
+		try {
+			shellStartCommand = await which('pwsh.exe')
+			return shellStartCommand
+		} catch (e1) {
+			try {
+				shellStartCommand = await which('powershell.exe')
+				return shellStartCommand
+			} catch (e2) {
+				shellStartCommand = process.env.COMSPEC || 'cmd.exe'
+				return shellStartCommand
+			}
+		}
+	} else {
+		// Unix
+		shellStartCommand = process.env.SHELL || '/bin/sh'
+		return shellStartCommand
+	}
+}
+
+// set start command asyncronously
+export async function setShellStartCommand () {
+	const shellStartCommand = await getShellStartCommand()
+	atom.config.set('x-terminal.spawnPtySettings.command', shellStartCommand)
```

Comment:
If someone changes this setting it will be overwritten the next time they start Atom

Reply:
This is fixed in https://github.com/bus-stop/x-terminal/pull/128/commits/11c8dd7ff7a6de7f18f5701d811c1e28fcefb66d